lib/Runtime/Library/AtomicsObject.cpp

```diff
@@ -53,23 +53,27 @@ namespace Js
 
     uint32 AtomicsObject::ValidateAtomicAccess(Var typedArray, Var requestIndex, ScriptContext *scriptContext)
     {
+        Assert(TypedArrayBase::Is(typedArray));
+
         int32 accessIndex = -1;
         if (TaggedInt::Is(requestIndex))
```

Comment:
we should make use of ArrayBuffer::ToIndex. That function should do the exactly the same.